src/pipx/commands/commands.py

```diff
@@ -60,18 +70,47 @@ def install(
         except PackageInstallFailureError:
             venv.remove_venv()
             raise PipxError(
-                f"Could not install package {package}. Is the name or spec correct?"
+                f"Could not install package {package_name}. Is the name or spec correct?"
             )
 
         _run_post_install_actions(
-            venv, package, local_bin_dir, venv_dir, include_dependencies, force=force
+            venv,
+            package_name,
+            local_bin_dir,
+            venv_dir,
+            include_dependencies,
+            force=force,
         )
     except (Exception, KeyboardInterrupt):
         print("")
         venv.remove_venv()
         raise
 
 
+def _package_name_from_spec(package_spec: str, python: str) -> str:
+    start_time = time.time()
+
+    package_name: Optional[str]
+
+    # shortcut if valid PYPI name and not a local path
+    if valid_pypi_name(package_spec) and not Path(package_spec).exists():
+        package_name = package_spec
+        logging.info(f"Determined package name: {package_name}")
+        logging.info(f"Package name determined in {time.time()-start_time:.1f}s")
+        return package_name
+
+    with tempfile.TemporaryDirectory() as temp_venv_dir:
+        venv = Venv(Path(temp_venv_dir), python=python)
+        venv.create_venv(venv_args=[], pip_args=[])
+        package_name = venv.install_package_no_deps(
+            package_or_url=package_spec, pip_args=[]
+        )
+        if package_name is None:
+            raise PipxError(f"Unable to validate {package_spec}")
```

Comment:
I just ran into this locally while testing. 
```
>> pipx install gdbgui
Unable to validate gdbgui
```
This error message is technically correct, but I think a little more context and next steps would be helpful. Maybe something like
```
raise PipxError(f"Package was found and installed, but it is invalid. Verify the package  {package_spec} can be installed with pip and then try again.
```

Reply:
I'm going to have to investigate that, because it shouldn't be getting to that error message because gdbgui is a valid pypi name and presumably not a local path for your.